File: tests/fixtures/sample_projects/sample_project_swift/Shapes.swift
import Foundation

// Protocol for shapes
protocol Shape {
    func area() -> Double
    func perimeter() -> Double
}

// Circle class implementing Shape
class Circle: Shape {
    let radius: Double
    
    init(radius: Double) {
        self.radius = radius
    }
    
    func area() -> Double {
        return Double.pi * radius * radius
    }
    
    func perimeter() -> Double {
        return 2 * Double.pi * radius
    }
}

// Rectangle struct implementing Shape
struct Rectangle: Shape {
    let width: Double
    let height: Double
    
    func area() -> Double {
        return width * height
    }
    
    func perimeter() -> Double {
        return 2 * (width + height)
    }
    
    func isSquare() -> Bool {
        return width == height
    }
}

// Triangle class implementing Shape
class Triangle: Shape {
    let base: Double
    let height: Double
    let sideA: Double
    let sideB: Double
    
    init(base: Double, height: Double, sideA: Double, sideB: Double) {
        self.base = base
        self.height = height
        self.sideA = sideA
        self.sideB = sideB
    }
    
    func area() -> Double {
        return 0.5 * base * height
    }
    
    func perimeter() -> Double {
        return base + sideA + sideB
    }
}

File: tests/fixtures/sample_projects/sample_project_scala/Shapes.scala
package com.example.shapes

sealed trait Shape {
  def area: Double
  def perimeter: Double
}

case class Circle(radius: Double) extends Shape {
  override def area: Double = Math.PI * radius * radius
  override def perimeter: Double = 2 * Math.PI * radius
}

case class Rectangle(width: Double, height: Double) extends Shape {
  override def area: Double = width * height
  override def perimeter: Double = 2 * (width + height)
}

object ShapeFactory {
  def createCircle(r: Double): Shape = Circle(r)
  def createRectangle(w: Double, h: Double): Shape = Rectangle(w, h)
}
